### ISSUE TITLE
Incorrect targetStore for UndoManager when set as a property in a child model

### ISSUE THREAD

**Issue by robclouth** [mobxjs/mobx-state-tree#832]
Hey,
I'm creating the undoManager in the Todo object like this:
```
types
  .model("Todo", {
    id: types.identifier(),
    position: types.optional(types.frozen, { x: 0, y: 0 }),
    history: types.optional(UndoManager, {})
  })
```
Then I add a Todo to the main store and call undo like this:
```
store.createTodo({ id: "0" });
store.todos.get("0").setPosition({ x: 10, y: 10 });
store.todos.get("0").history.undo();
```
But I get this error:
```
Not a value property: todos
```
It seems like the UndoManager is attaching to the parent store instead of the Todo. 
I can get it to attach to the Todo if I create it like 
```
UndoManager.create({}, { targetStore })
```
But I'd like to save the undo history in the Todo model for saving and loading purposes.

https://codesandbox.io/s/4qp264rw1w

**Comment by robinfehr** [comment]
first: please update the packages :)

easy solution: 
add the history on the parent.

harder solution:
you could have a quick look at the `mst-middlewares` package code and search for 
```js
        return {
            recorder: mst.recordPatches(call.tree),
            actionId: actionId
        };
```
we might need to change the call.tree...
but not sure that works you'd have to run the tests afterwards...
sorry currently not having much time.

**Comment by robclouth** [comment]
Ok I've figured it out:
```
return {
   recorder: mst.recordPatches(call.tree),
   actionId: actionId
};
```
call.tree should be call.**context**

Also in afterCreate:
```
targetStore = mst.getEnv(self).targetStore ? mst.getEnv(self).targetStore : mst.getRoot(self);
```
mst.getRoot(self) should be mst.**getParent**(self)

I've submitted a PR. The tests pass.

**Comment by mweststrate** [comment]
Should be fixed in 2.2.0
